OrbitCore/TracepointEventBuffer.cpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "TracepointEventBuffer.h"
+
+using orbit_client_protos::TracepointEventInfo;
+
+std::vector<TracepointEventInfo> TracepointEventBuffer::GetTracepointEvents(
+    uint64_t time_begin, uint64_t time_end, int32_t thread_id /*= kAllThreadsFakeTid*/) {
+  ScopeLock lock(mutex_);
+  std::vector<TracepointEventInfo> tracepoint_events;
+  for (auto& pair : tracepoint_events_) {
+    const int32_t tracepoint_thread_id = pair.first;
+    std::map<uint64_t, TracepointEventInfo>& tracepoints = pair.second;
+
+    CHECK(thread_id == SamplingProfiler::kAllThreadsFakeTid || tracepoint_thread_id == thread_id);
+    for (auto it = tracepoints.lower_bound(time_begin); it != tracepoints.end(); ++it) {
+      uint64_t time = it->first;
+      if (time < time_end) {
+        tracepoint_events.push_back(it->second);
+      } else
+        return tracepoint_events;
```

Comment:
Shouldn't this be a `break` and then you continue with the next iteration of the outer loop? If `thread_id == SamplingProfiler::kAllThreadsFakeTid`, you want to always go through all elements in `tracepoint_events_`.

Reply:
Yes, you are right. Thank you!